### DISCUSSION TITLE
Setup my own smtp server

### DISCUSSION THREAD

**Discussion by papapon** [fief-dev/fief#12] [Q&A]
Hello, 

I came across this project using fastapi-user. Your work is great. So I wanted to look at fief. It looks great too. 

However I don't want to use postmark at all. I would like to use my own smtp server. I saw the option : 
EMAIL_PROVIDER=NULL in the documentation. So I think that it is possible to configure my smtp server.

Is it possible? or it's another feature to implement?

**Comment by frankie567** [comment]
Hi @papapon! Welcome to Fief!

Unfortunately, there is no class implemented for generic SMTP server right now, but we plan to implement more providers soon. The `NULL` provider is a mock provider that doesn't send emails.

If you or anybody else want to help with this, I would happily welcome a PR with an SMTP provider implementation. Basically, it involves three things:

* Implement a class for the new provider inheriting from [`BaseProvider`](https://github.com/fief-dev/fief/blob/cbfeec11da7a03aa345cb7ceb088b5d8ec9d6ab1/backend/fief/services/email/base.py)
* Add it to the [`AvailableEmailProvider` enum](https://github.com/fief-dev/fief/blob/cbfeec11da7a03aa345cb7ceb088b5d8ec9d6ab1/backend/fief/services/email/__init__.py#L9-L11) and to the [`EMAIL_PROVIDERS` mapping](https://github.com/fief-dev/fief/blob/cbfeec11da7a03aa345cb7ceb088b5d8ec9d6ab1/backend/fief/services/email/__init__.py#L14-L17)
* Add documentation about it in the dedicated page in the docs: https://github.com/fief-dev/docs/blob/main/docs/self-hosting/setup-email-provider.md
